src/state/slices/portfolioSlice/selectors.ts

```diff
@@ -322,10 +322,30 @@ export const selectPortfolioAllocationPercentByFilter = createSelector(
 
 export const selectPortfolioAccountIdsSortedFiat = createSelector(
   selectPortfolioTotalFiatBalanceByAccount,
-  totalAccountBalances => {
-    return Object.entries(totalAccountBalances)
-      .sort(([_, a], [__, b]) => (bnOrZero(a).gte(bnOrZero(b)) ? -1 : 1))
-      .map(([acctId, _]) => acctId)
+  selectAssets,
+  (totalAccountBalances, assets) => {
+    const sortedAccountBalances = Object.entries(totalAccountBalances)
+      .sort(([_, accountBalanceA], [__, accountBalanceB]) =>
+        bnOrZero(accountBalanceA).gte(bnOrZero(accountBalanceB)) ? -1 : 1
+      )
+      .map(([accountId, _]) => accountId)
+
+    const sortedAccountBalancesByChainBuckets = sortedAccountBalances.reduce(
+      (acc: Record<ChainTypes, CAIP10[]>, accountId) => {
+        const assetId = accountIdToFeeAssetId(accountId)
+        const asset = assets[assetId]
+
+        if (!acc[asset.chain]) {
+          acc[asset.chain] = []
+        }
+
+        acc[asset.chain] = [...acc[asset.chain], accountId]
+        return acc
+      },
+      {} as Record<ChainTypes, CAIP10[]>
```

Comment:
Note: Agreed with @majorhayes to abstract this in their own functions as a follow-up PR and test them